File: app/src/main/java/com/nfcalarmclock/statistics/NacAlarmMissedStatistic.java
package com.nfcalarmclock.statistics;

import androidx.room.Entity;
import androidx.room.ForeignKey;

import com.nfcalarmclock.alarm.NacAlarm;

/**
 * Statistics for when an alarm is missed.
 */
@Entity(tableName="alarm_missed_statistic",
	foreignKeys={
		@ForeignKey(entity=NacAlarm.class,
			parentColumns={"id"},
			childColumns={"alarm_id"},
			onDelete=ForeignKey.SET_NULL)
		})
public class NacAlarmMissedStatistic
	extends NacAlarmStatistic
{

	/**
	 */
	public NacAlarmMissedStatistic()
	{
		super();
	}

	/**
	 */
	public NacAlarmMissedStatistic(NacAlarm alarm)
	{
		super(alarm);
	}

}

File: app/src/main/kotlin/com/nfcalarmclock/statistics/db/NacAlarmMissedStatistic.kt
package com.nfcalarmclock.statistics.db

import androidx.room.Entity
import com.nfcalarmclock.alarm.db.NacAlarm
import dagger.Module
import dagger.Provides
import dagger.hilt.InstallIn
import dagger.hilt.components.SingletonComponent

/**
 * Statistics for when an alarm is missed.
 */
@Entity(tableName = "alarm_missed_statistic")
class NacAlarmMissedStatistic
	: NacAlarmStatistic
{

	/**
	 * Constructor.
	 */
	constructor() : super()

	/**
	 * Constructor.
	 */
	constructor(alarm: NacAlarm?) : super(alarm)

}

/**
 * Hilt module to provide an instance of a missed statistic.
 */
@InstallIn(SingletonComponent::class)
@Module
class NacAlarmMissedStatisticModule
{

	/**
	 * Provide an instance of a missed statistic.
	 */
	@Provides
	fun provideMissedStatistic() : NacAlarmMissedStatistic
	{
		return NacAlarmMissedStatistic()
	}

}
